Disable nix tooling

gngram opened the pull request:
<!--
    Copyright 2023 TII (SSRC) and the Ghaf contributors
    SPDX-License-Identifier: CC-BY-SA-4.0
-->

## Description of changes

<!--
Summary of the proposed changes in the PR description in your own words. For dependency updates, please link to the changelog.
-->
- Disables nix tooling if development setup is not enabled
         - Indirectly disables nix tooling in release build
- Enables automatic garbaze collection in ghaf-host and disable everywhere else.
- Fixes:
    https://jira.tii.ae/browse/SSRCSP-3457

## Checklist for things done

<!-- Please check, [X], to all that applies. Leave [ ] if an item does not apply but you have considered the check list item. Note that all of these are not hard requirements. They serve information to reviewers. When you fill the checklist, you indicate to reviewers you appreciate their work. -->

- [x] Summary of the proposed changes in the PR description
- [x] More detailed description in the commit message(s)
- [x] Commits are squashed into relevant entities - avoid a lot of minimal dev time commits in the PR
- [ ] [Contribution guidelines](https://github.com/tiiuae/ghaf/blob/main/CONTRIBUTING.md) followed
- [ ] Ghaf documentation updated with the commit - https://tiiuae.github.io/ghaf/
- [ ] PR linked to architecture documentation and requirement(s) (ticket id)
- [ ] Test procedure described (or includes tests). Select one or more:
  - [x] Tested on Lenovo X1 `x86_64`
  - [ ] Tested on Jetson Orin NX or AGX `aarch64`
  - [ ] Tested on Polarfire `riscv64`
- [ ] Author has run `make-checks` and it passes
- [ ] All automatic Github Action checks pass - see [actions](https://github.com/tiiuae/ghaf/actions)
- [ ] Author has added reviewers and removed PR draft status
- [ ] Change requires full re-installation
- [ ] Change can be updated with `nixos-rebuild ... switch`

<!-- Additional description of omitted [ ] items if not obvious. -->

## Instructions for Testing

- [x] List all targets that this applies to:
     - Lenovo X1
     - Jetson Orin NX and AGX 
- [ ] Is this a new feature
  - [ ] List the test steps to verify:
      - In release build, no nix commands should be available, nix daemon should not be running.
            - As Terminal is not available in release flavor, disable nix-setup in `./modules/common/profiles/debug.nix` (to imitate release)
               ```
               nix-setup.enable = false;
               ```
           - Build debug image and run. No nix commands should be available, nix daemon should not be running. 
                
      - In debug build nix daemon should be running in all VMs and automatic garbage collection should be available only in host.
      - If automatic garbaze collection is enabled 'systemctl list-timers' should list 'nix-gc' also
- [ ] If it is an improvement how does it impact existing functionality?

gngram commented on modules/common/development/nix.nix:
```diff
@@ -18,37 +19,49 @@ in
-  };
+    })
+
+    (mkIf (!cfg.enable) {
```
Done!

gngram replied:
Rebased to latest.

gngram commented:
Rebased to latest.

gngram commented on modules/common/development/nix.nix:
```diff
@@ -18,37 +19,49 @@ in
       description = "Path to the nixpkgs repository";
     };
+    automatic-gc = {
+      enable = mkEnableOption "Enable automatic garbaze collection";
```
Done.

leivos-unikie commented:
Tested debug and release disk image on Lenovo-X1

Lenovo-X1 release image
- There is no access to terminal / command line. Terminal is not available in the app menu, neither can it be opened via right mouse click. Ssh to ghaf-host is disabled. I didn't find any way to verify if nix tooling and nix daemon are disabled.

Lenovo-X1 debug image
- nix-gc.timer is active in all VMs, contrary to Intructions for Testing
```
NEXT                          LEFT LAST                          PASSED UNIT                         ACTIVATES
Wed 2025-01-08 00:00:00 UTC    14h -                                      - nix-gc.timer                 nix-gc.service
```
- nix tooling seems to be available in every VM but `systemctl status nix-daemon` shows inactive (dead) status (unless running first e.g. `nix-store --gc` in that VM).
- ci-test-automation passes, performance ok

leivos-unikie commented:
Tested that release-installer succeeds to install ghaf release version.

johannarautanen commented:
**Tested debug and release disk image on Orin AGX**

**Orin AGX release image:**
- There is no access to terminal / command line. Terminal is not available in the app menu.

Tried to create the release image according the https://ghaf-jenkins-controller-prod.northeurope.cloudapp.azure.com/job/ghaf-release-pipeline/2/ from October 2024, but there was no terminal available either. Rumors told that last time the terminal was seen on release image in summer 2024.

**Orin AGX debug image:**

- nix-gc.timer is active in all VMs; host and net-vm
![Screenshot from 2025-01-08 11-11-39](https://github.com/user-attachments/assets/9873fc77-e646-48c7-9778-00d7a7454b64)

- Like in Lenovo-X1 the nix tooling seems to be available in every VM but systemctl status nix-daemon shows inactive (dead) status (unless running first e.g. nix-store --gc in that VM).

- test automation cases were ok

gngram commented:
@johannarautanen and @leivos-unikie , please test it again. I have re-based it and fixed the issue.
I have updated the test steps also to imitate release flavor.

leivos-unikie commented:
Tested again on lenovo-x1

Debug image built with the change (nix-setup disabled) imitating release image
- nix tooling not available in ghaf-host or any VM
- nix-daemon.service not found in ghaf-host or any VM

Debug image
- nix-gc.timer is active only on ghaf-host
- nix tooling available in ghaf-host and in all VMs
- although nix tooling is available in ghaf-host and in every VM systemctl status nix-daemon shows inactive (dead) by default

johannarautanen commented:
**Checked with Orin AGX**
Imitate release (nix-setup.enable = false;)
- nix-tooling not available in host or net-vm
- nix-daemon.service not found in ghaf-host 
When checking with the `systemctl status nix-daemon` in net-vm:
![Screenshot from 2025-01-21 12-52-27](https://github.com/user-attachments/assets/ccaacd7a-a169-472f-837f-2b1cfd49383b)

Debug:
- nix-gc.timer is active only visible in ghaf-host
- nix tooling available in ghaf-host and in all VMs
- And same as in x1: although nix tooling is available in ghaf-host and in every VM systemctl status nix-daemon shows inactive (dead) by default

johannarautanen commented:
**Checked with Orin NX**

Imitate release (nix-setup.enable = false;)
- nix-tooling not available in host or net-vm
- nix-daemon.service not found in ghaf-host
Same as with AGX, when checking with the systemctl status nix-daemon in net-vm:
![Screenshot from 2025-01-21 12-52-27](https://github.com/user-attachments/assets/719bf47c-53b3-475e-aaef-e33561eb9234)

Debug:
- nix-gc.timer is only visible in ghaf-host
- nix tooling available in ghaf-host and net-vm
And same as in x1: although nix tooling is available in ghaf-host and and net-vm status nix-daemon shows inactive (dead) by default

ktusawrk commented:
Failed ghaf-pre-merge-pipeline builds 326 and 329 are not related to this PR. There was a glitch in the pre-merge-pipeline itself. These failed builds do not prevent merging this PR.

gngram commented:
Rebased
